libr/core/cmd_print.c

```diff
@@ -3333,6 +3333,67 @@ static bool cmd_print_blocks(RCore *core, const char *input) {
 	return result;
 }
 
+static bool cmd_pse(RCore *core, const char *input) {
+	if (input[2] == ' ') {
+		const char *fname = r_str_trim_head_ro (input + 3);
+		char *tmp = r_str_newf (R_JOIN_2_PATHS (R_JOIN_2_PATHS (R2_HOME_DATADIR, "util"), "%s"), fname);//todo: add R2_HOME_UTIL
```

Comment:
it should be possible to embed those encodings inside r2 or maybe in the futur will be another type of encoder plugins.

Move all this logic of walking the system directories into a function that updates the charser_db or returns false

Reply:
@trufae Do you mean in radare2/libr/include/r_userconf.h ?

Reply:
The code in this function should be something like this (in pseudocode):

```c
char *s = read_string_from_current_offset();
char *o = r_util_encode (s, encoder);
r_cons_printf ("%s", o);
free (s);
free (o);
```

The way to select one encoder/decoder should be via cfg.something, because thats something that shouldnt change frrequently. so you can keep that configuration in core and decode multiuple strings later by using the pse command.

Note the string is not passed as argument, but instead read from current offset, ideally the argument to pse should be a number, to override the core->blocksize

Reply:
please ^